src/Assignments/SimpleProblemPage.tsx

```diff
@@ -379,12 +418,15 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
                                                 New Version
                                             </Button>
                                         }
-                                        <Button variant='danger'
-                                            tabIndex={0}
-                                            onClick={() => confirmEndVersion()}
-                                        >
-                                            End Exam
-                                        </Button>
+                                        {(attemptsRemaining!==topic.topicAssessmentInfo?.maxGradedAttemptsPerVersion) &&
+                                            <Button variant='danger'
+                                                tabIndex={0}
+                                                onClick={() => confirmEndVersion()}
```

Comment:
No need to pass a function that calls a function
```
onClick={confirmEndVersion}
```

Reply:
I do, because confirmEndVersion has optional arguments.